var/spack/repos/builtin/packages/py-grpcio/package.py

```diff
@@ -15,6 +15,7 @@ class PyGrpcio(PythonPackage):
     license("Apache-2.0")
 
     version("1.60.1", sha256="dd1d3a8d1d2e50ad9b59e10aa7f07c7d1be2b367f3f2d33c5fade96ed5460962")
+    version("1.56.2", sha256="0ff789ae7d8ddd76d2ac02e7d13bfef6fc4928ac01e1dcaa182be51b6bcc0aaa")
```

Comment:
Updated dependencies back to 1.35.0. Should we consider deprecating those before that?

Reply:
Added deps in https://github.com/spack/spack/pull/44263/commits/5e6c1d2001709f284dc878cd02d61f45dc1bc752

Reply:
Good question about deprecating older versions.  I see on the order of ten packages using it with one (`py-tensorflow`) going back as far as `1.8.6` with versions that aren't deprecated.

What do you think @adamjstewart ?

Reply:
The decision whether or not to deprecate old versions of packages is up to the maintainer of that recipe. Since there currently is none, that means you can do whatever you want and no one can complain (unless of course they want to volunteer to maintain it themselves).

For Python, I follow [PSF's support schedule](https://devguide.python.org/versions/) (5 years of maintenance). For most other packages (including TensorFlow), I deprecate versions that only support deprecated versions of Python. Which in effect means a 5 year schedule for those too. [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html) recommended 3.5 years of maintenance for scientific Python software. [SPEC 0](https://scientific-python.org/specs/spec-0000/) replaces that and recommends 3 years instead.

I usually get yelled at when I try to remove ancient versions of stuff, so other than reasons like "these ancient versions literally cannot be compiled with any supported Python version", I tend to only deprecate stuff when it will allow us to simplify the package recipe. 

In the case of py-grpcio, there is no additional package complexity (no code is specific to older versions). So unless older versions can't be built with Python 3.8 (the current oldest supported Python), I would suggest leaving them alone. But again, it's up to you.

Hope this helps!

Reply:
Oh, one other thing. I like to see if these older versions are required by any other packages before deprecating. In the case of py-grpcio, we have:
```console
> spack pkg grep 'depends_on..py-grpcio@.*:[0-9]'
/Users/Adam/spack/var/spack/repos/builtin/packages/py-apache-beam/package.py:    depends_on("py-grpcio@1.29:1", type=("build", "run"))
/Users/Adam/spack/var/spack/repos/builtin/packages/py-google-api-core/package.py:            depends_on("py-grpcio@1.49.1:1", when="^python@3.11:", type="run")
/Users/Adam/spack/var/spack/repos/builtin/packages/py-google-api-core/package.py:            depends_on("py-grpcio@1.33.2:1", type="run")
/Users/Adam/spack/var/spack/repos/builtin/packages/py-google-api-core/package.py:            depends_on("py-grpcio@1.33.2:1", when="@1.33:", type="run")
/Users/Adam/spack/var/spack/repos/builtin/packages/py-google-api-core/package.py:            depends_on("py-grpcio@1.29.0:1", when="@1.19.1", type="run")
/Users/Adam/spack/var/spack/repos/builtin/packages/py-google-api-core/package.py:            depends_on("py-grpcio@1.8.2:1", type="run")
/Users/Adam/spack/var/spack/repos/builtin/packages/py-googleapis-common-protos/package.py:        depends_on("py-grpcio@1.44:1", when="@1.57:", type="run")
/Users/Adam/spack/var/spack/repos/builtin/packages/py-grpc-google-iam-v1/package.py:    depends_on("py-grpcio@1.44:1", type=("build", "run"))
/Users/Adam/spack/var/spack/repos/builtin/packages/py-grpcio-tools/package.py:    depends_on("py-grpcio@1.39.0:", when="@1.39.0:1.41", type=("build", "run"))
/Users/Adam/spack/var/spack/repos/builtin/packages/py-ray/package.py:    depends_on("py-grpcio@1.32:1.43.0", when="@2.0.1 ^python@:3.9", type=("build", "run"))
/Users/Adam/spack/var/spack/repos/builtin/packages/py-ray/package.py:    depends_on("py-grpcio@1.42:1.43.0", when="@2.0.1 ^python@3.10:", type=("build", "run"))
/Users/Adam/spack/var/spack/repos/builtin/packages/py-tensorflow/package.py:        depends_on("py-grpcio@1.24.3:1", type=("build", "run"), when="@2.7:")
/Users/Adam/spack/var/spack/repos/builtin/packages/py-tensorflow/package.py:        depends_on("py-grpcio@1.37.0:1", type=("build", "run"), when="@2.6")
/Users/Adam/spack/var/spack/repos/builtin/packages/py-tensorflow/package.py:        depends_on("py-grpcio@1.8.6:", type=("build", "run"), when="@:2.3")
```
So py-ray and py-grpcio-tools require py-grpcio 1.43, but everything else doesn't really care.

Reply:
This is very helpful! I was wondering what the guidance and standard practices are. Also the `spack pkg grep` formula is very helpful. I'm inclined to put in the effort to properly document the dependencies going back further as this seems to be one of those packages where downstream packages can have specific version needs. 

I don't have very much experience with this package as well but I may put some time in to figure out how to properly test it for these older versions.

I suspect `py-grpcio`  is could also use some updates and is pinned to versions (like `py-grpcio-status` is).

Reply:
It seems we should deprecate two of these versions: 1.52.0 was yanked from pypi and the other doesn't install with python3.8. It may be that my lowest available GCC versions is too high, but the version was for <=python3.6.